core/src/main/java/com/digitalpebble/stormcrawler/util/MetadataTransfer.java

```diff
@@ -156,12 +156,34 @@ public Metadata filter(Metadata metadata) {
         return filtered_md;
     }
 
+    /**
+     * Filter the metadata based on a set of keys. If a key ends with a * then all the keys starting
+     * with the prefix will be added.
+     */
     private Metadata _filter(Metadata metadata, Set<String> filter) {
         Metadata filtered_md = new Metadata();
+
+        Set<String> filterKeys = new HashSet<>();
         for (String key : filter) {
-            String[] vals = metadata.getValues(key);
-            if (vals != null) filtered_md.setValues(key, vals);
+            if (key.endsWith("*")) {
+                String prefix = key.substring(0, key.length() - 1);
+                for (String mdKey : metadata.keySet()) {
```

Comment:
Could add a new method to Metadata

_ Set<String> getKeysWithPrefix(String prefix)_

as this is certainly used elsewhere

and iterate on it e.g.

  for (String mdKey : metadata.getKeysWithPrefix(prefix)) {

Reply:
Thank you for the hint. I named the new method `keySet(String prefix)` for keeping the same naming schema as `metadata.keySet()`. It is indeed also used in FetcherBolt and SimpleFetcherBolt.